redbox-review/redbox.go

```diff
@@ -0,0 +1,293 @@
+package redbox
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cgclever/redbox/s3box"
+)
+
+const defaultNManifests = 4
+
+var (
+	errShippingInProgress = fmt.Errorf("Cannot perform any action when shipping is in progress.")
+	errIncompleteArgs     = fmt.Errorf("Creating a redshift box requires a schema, table and an s3 bucket.")
+	errInvalidJSONInput   = fmt.Errorf("Only JSON inputs are supported.")
+	errBoxShipped         = fmt.Errorf("Cannot perform any actions, the box has been shipped.")
+	errNothingToShip      = fmt.Errorf("Cannot perform send, no data was packed.")
+)
+
+// Redbox manages piping data into Redshift.
+// An S3Box is used to manage data transport to S3 via Pack, after
+// which Ship commits the data to Redshift.
+type Redbox struct {
```

Comment:
Good question. I did this for the sake of terser function calls, making the configuration more verbose than the methods. However, this might be solved by making the struct value
```
type Redbox struct {
  o NewRedboxOptions
}

func (rb *Redbox) someMethod() {
  schema := rb.o.Schema
}
```
Thoughts?